src/evm/protocol/vm/adapter_contract.rs

```diff
@@ -47,197 +55,144 @@ where
 {
     pub fn price(
         &self,
-        pair_id: Vec<u8>,
+        pair_id: String,
```

Comment:
This gets immediately converted to a reference and used only as a reference. Why not pass it as a reference here already and avoid needing to clone the pair_id at the call site?